content/docs/rfcs/59/README.md

```diff
@@ -0,0 +1,129 @@
+---
+slug: 59
+title: 59/STATUS-URL-DATA
+name: Status URL Data
+status: raw
+category: Standards Track
+tags: waku-application
+editor: Felicio Mununga <felicio@status.im>
+contributors:
+---
+
+# Abstract
+
+This document presents a comprehensive overview of the serialization, compression, and encoding techniques used to transmit data within URLs. 
+The primary objective is to enable app links to contain sufficient information, allowing users to conveniently preview and verify the content before taking any actions or clicking on the link. 
+By employing these methods, users can make informed decisions about engaging with the content, enhancing their overall browsing experience.
+
+# Background / Rationale / Motivation
+
+## Requirements
+
+### Product
+
+- The process of encoding data within URLs should continue until the Waku response becomes sufficiently instantaneous and reliable for link previewing and visiting the onboarding website. Alternatively, this process can be halted once a product roadmap for an alternative decentralized storage solution is finalized.
+
+- During the loading of the onboarding website, a verification state should be implemented to compare the encoded data against the Waku response. This measure aims to mitigate identity attacks and ensure the integrity of the data.
+
+- Only characters from the [A-Za-z0-9_-.\u0020] character class should be accepted for textual fields. This restriction promotes compatibility and security in the handling of textual data.
+
+- Strive for the shortest possible result, optimizing the encoding process to minimize the overall length of the encoded data while maintaining its intended functionality and integrity.
+### Related scope
+
+#### Features
+
+- Onboarding website
+- Link preview
+- Link sharing
+- Deep linking
+- Routing and navigation
+
+#### App entities
+
+- Community: Refer to [56/STATUS-COMMUNITIES](/spec/56)
+- Channel: Refer to terminology in [56/STATUS-COMMUNITIES](/spec/56)
+- User: Refer to terminology in [56/STATUS-COMMUNITIES](/spec/56)
+
+# Wire Format Specification / Syntax
+
+## Data
+
+### Fields
+
+```protobuf
+syntax = "proto3";
+
+message URLData {
+ // Community, Channel, or User
+ bytes content = 1;
+}
+
+// Field on CommunityDescription, CommunityChat and ContactCodeAdvertisement (described in 56/STATUS-COMMUNITIES)
+message URLParams {
+ string encoded_url_data = 1;
+ // Signature of encoded URL data
+ string encoded_signature = 2;
+}
+```
+
+# Implementation Suggestions
+
+## Example
+
+- See <https://github.com/status-im/status-web/pull/345/files>
+
+## Encoding
+
+- Base64url
```

Comment:
It was under "suggestion" due to the "request for comments" and "draft" or "raw" state nature. Especially, since I got very little technical feedback on the whole spec.

Reply:
Matter of fact, and spec contributing/format aside, what do you think about the (suggested) implementation? Would solve the problem differently?